js/service/shell/index.js

```diff
@@ -20,6 +20,41 @@ var runtime = require('../../core');
 var tty = runtime.tty;
 var commands = new Map();
 
+exports.setCommand = function(name, cb) {
+  assert(typeutils.isString(name));
+  assert(typeutils.isFunction(cb));
+  commands.set(name, cb);
+};
+
+exports.runCommand = function(name, args, done) {
+  var opts = {};
+
+  assert(typeutils.isString(name));
+
+  if (typeutils.isArray(args)) {
+    opts.args = args;
+  } else {
+    opts = args;
+    opts.args = [];
+  }
+
+  var stringargs = opts.args.join(' ');
+
+  opts.stdout = opts.stdout || env.stdout;
+  opts.stdin = opts.stdin || env.stdin;
+
+  try {
```

Comment:
try catch wouldn't be able to catch async errors, so might be confusing that sometimes it works (error in sync code) and sometimes not (error in async code). I'd remove it to make it behave the same way